packages/@aws-cdk/aws-codebuild/README.md

```diff
@@ -119,3 +124,103 @@ To define CloudWatch event rules for build projects, use one of the `onXxx` meth
 const rule = project.onStateChange('BuildStateChange');
 rule.addTarget(lambdaFunction);
 ```
+
+### Secondary sources and artifacts
+
+CodeBuild Project can get their sources from multiple places,
+and produce multiple outputs. For example:
+
+```ts
+const project = new codebuild.Project(this, 'MyProject', {
+    secondarySources: [
+        new codebuild.CodeCommitSource({
+            identifier: 'source2',
+            repository: repo,
+        }),
+    ]
+    secondaryArtifacts: [
+        new codebuild.S3BucketBuildArtifacts({
+            identifier: 'artifact2',
+            bucket: bucket,
+            path: 'some/path',
+            name: 'file.zip',
+        }),
+    ],
+    // ...
+});
+```
+
+The `identifier` property is required for both secondary sources and artifacts.
+
+The contents of the secondary source will be available to the build under the directory
+specified by the `CODEBUILD_SRC_DIR_<identifier>` environment variable
+(so, `CODEBUILD_SRC_DIR_source2` in the above case).
+
+The secondary artifacts have their own section in the buildspec,
+under the regular `artifacts` one.
+Each secondary artifact has its own section,
+beginning with their identifier.
+
+So, a buildspec for the above Project could look something like this:
+
+```ts
+const project = new codebuild.Project(this, 'MyProject', {
+    // secondary sources and artifacts as above...
+    buildSpec: {
+        version: '0.2',
+        phases: {
+            build: {
+                commands: [
+                    'cd $CODEBUILD_SRC_DIR_source2',
+                    'touch output2.txt',
+                ],
+            }
+        },
+        artifacts: {
+            'secondary-artifacts': {
```

Comment:
OMFG the backwards compatible API extensions they pain me so much 

![oh no](https://media.giphy.com/media/F0dSz5pqsyU3S/giphy.gif)

Reply:
Lol :)

Reply:
Just wait until CDK hits 1.0, and we'll have to do crap like this ourselves 😀